includes/Settings/SettingsDebug.php

```diff
@@ -1975,6 +1978,194 @@ public function maybe_schedule_unstable_version_check(): void {
 	public function run_unstable_version_check(): void {
 		wpo_wcpdf_get_latest_releases_from_github();
 	}
+	
+	/**
+	 * Generate and download the plugin report as a PDF.
+	 *
+	 * @return void
+	 */
+	public function ajax_plugin_report(): void {
+		if ( ! \WPO_WCPDF()->settings->user_can_manage_settings() ) {
+			wp_die( __( 'You are not allowed to perform this action.', 'woocommerce-pdf-invoices-packing-slips' ) );
+		}
+
+		check_ajax_referer( 'wpo_ips_plugin_report', 'nonce' );
+		
+		$include_sensitive    = isset( $_GET['include_sensitive'] )
+			? filter_var( wp_unslash( $_GET['include_sensitive'] ), FILTER_VALIDATE_BOOLEAN )
+			: false;
+	
+		$report_title         = 'PDF Invoices & Packing Slips for WooCommerce - Report';
+		$premium_plugins      = $this->get_premium_plugins();
+		$free_extensions      = $this->get_free_extensions();
+		$multilingual_plugins = $this->get_multilingual_plugins();
+		$plugin_version       = \WPO_WCPDF()->version;
+		$store_url            = get_site_url();
+		$report_date          = gmdate( 'Y-m-d H:i:s' );
+		$report_user          = wp_get_current_user();
+		$server_configs       = $this->get_server_config();
+		$dir_permissions      = $this->get_directory_permissions();
+		$yearly_reset         = $this->get_yearly_reset_schedule();
+		
+		// Load CSS file contents
+		$report_css = '';
+		$suffix     = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+		$css_path   = \WPO_WCPDF()->plugin_path() . '/assets/css/plugin-report' . $suffix . '.css';
+
+		if ( \WPO_WCPDF()->file_system->exists( $css_path ) && \WPO_WCPDF()->file_system->is_readable( $css_path ) ) {
+			$report_css = \WPO_WCPDF()->file_system->get_contents( $css_path );
+		}
+		
+		// Load Settings
+		$general_settings   = get_option( 'wpo_wcpdf_settings_general', array() );
+		$debug_settings     = get_option( 'wpo_wcpdf_settings_debug', array() );
+		$edi_settings       = get_option( 'wpo_ips_edi_settings', array() );
+		$documents_settings = array();
+		
+		$all_documents = \WPO_WCPDF()->documents->get_documents( 'all' );
+		if ( ! empty( $all_documents ) ) {
+			foreach ( $all_documents as $document ) {
+				$document_type                        = $document->get_type();
+				$documents_settings[ $document_type ] = \WPO_WCPDF()->settings->get_document_settings( $document_type, 'pdf' );
+			}
+		}
+		
+		// Logs
+		$logs_data = $this->get_recent_logs();
+		
+		// Extensions Settings
+		$extensions_settings = apply_filters( 'wpo_ips_get_extensions_settings_for_report', array() );
+		
+		ob_start();
+		include \WPO_WCPDF()->plugin_path() . '/views/plugin-report.php';
+		$report_html = ob_get_clean();
+
+		$pdf_settings = array(
+			'paper_size'		=> 'A4',
+			'paper_orientation'	=> 'portrait',
+			'font_subsetting'	=> false,
+		);
+		$pdf_maker = \wcpdf_get_pdf_maker( $report_html, $pdf_settings, $this );
```

Comment:
The PDF worked perfectly in the 2 testing environments I tried, but I wonder what happen if `DOMDocument` is not available (I tried to disable it in Hostinger and Local to check, but it seems it comes enabled in the core). I think this is an edge case, but last week we have received a case like this ([see](https://secure.helpscout.net/conversation/3167153414/78008)).

Therefore, I was thinking that we could offer two options, like "Download report (PDF)" and "Download report (HTML)", just in case. What do you think?